### PR TITLE
tockloader: init at v1.4.0

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

Tockloader describes itself as a "Tool for programming Tock onto hardware boards."

Personally I'm trying to get OpenSK (https://github.com/google/OpenSK) running on a Nordic nRF52840 MDK and the deploy process requires among other things tockloader.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
How is the python-packages.nix top-level file sorted?

**Comment by jonringer** [comment]
> How is the python-packages.nix top-level file sorted?

it's not.

Some additions grouped similar packages. a  lot just did lexigraphical sorting.

Right now it's just a "best try"

**Comment by mweinelt** [comment]
Updated to disable python2 support, which was failing.

**Review by jonringer — COMMENTED** [review]
There are no tests included. Please try to checkout from source and check if they have unit tests, and try to run them. Unit tests give a good indication that they package has a high degree of validity and correctness given the python package set.

If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors.

```

----------------------------------------------------------------------
Ran 0 tests in 0.000s
```

**Comment by mweinelt** [comment]
I tried importing the TockLoader class, but couldn't properly adress it. I tried `tockloader.tockloader.TockLoader` (package.file.class).

https://github.com/tock/tockloader/blob/master/tockloader/tockloader.py#L28

So for now I settled for the package import, which isn't very helpful.

**Comment by mweinelt** [comment]
The package has no test suite and their CI job just tries to call `tockloader --version` and `tockloader list-known-boards`.

File: pkgs/development/python-modules/tockloader/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    pytoml
+    pyserial
+  ];
+
+  pythonImportsCheck = [
+    "tockloader"
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/tock/tockloader;
```
Comment (jonringer) [inline pkgs/development/python-modules/tockloader/default.nix:26]: rfc 45
```suggestion
    homepage = "https://github.com/tock/tockloader";
```

Comment (mweinelt) [reply pkgs/development/python-modules/tockloader/default.nix]: Done

**Comment by jonringer** [comment]
> The package has no test suite and their CI job just tries to call `tockloader --version` and `tockloader list-known-boards`.

That's.... concerning

**Review by jonringer — COMMENTED** [review]
is this only meant to be use as an application? (E.g `$ tockloader .... `)? If so, then it probably doesn't belong in `python-modules`. See https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonapplication-function

**Comment by mweinelt** [comment]
Ah yeah, that is probably the case. Thanks for the link.

**Comment by mweinelt** [comment]
Ok, ready for review again.

File: pkgs/development/tools/misc/tockloader/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    colorama
+    crcmod
+    pytoml
+    pyserial
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/tock/tockloader";
+    license = licenses.mit;
+    description = "Tool for programming Tock onto hardware boards.";
```
Comment (jonringer) [inline pkgs/development/tools/misc/tockloader/default.nix:23]: please add yourself as a maintainer :)

Comment (mweinelt) [reply pkgs/development/tools/misc/tockloader/default.nix:23]: Fair, done!

**Review by doronbehar — APPROVED** [review]
Works as far as `--help` is concerned.

**Pull request merged**